packages/workflow/src/records/fhir.ts

```diff
@@ -1377,6 +1390,32 @@ export function toSavedBundle<T extends Resource>(
         }
       }
 
+      if (
+        isTask(entry.resource) &&
+        entry.resource.extension &&
+        entry.resource.extension.some((x) => x.url.includes('paymentDetails'))
+      ) {
+        entry.resource.extension = entry.resource.extension.map((x: any) => {
```

Comment:
Can we create a completely new object here like in the other if branches above? Just noting this as we recently had a pretty nasty bug that was because of mutating

Reply:
okay